org.eclipse.xtext.common.types.ui/src/org/eclipse/xtext/common/types/xtext/ui/JdtHyperlinkFactory.java

```diff
@@ -34,7 +34,7 @@ public boolean canHandle(EObject to) {
 		return to instanceof JvmIdentifiableElement && to.eResource() instanceof TypeResource;
 	}
 
-	public void createHyperlink(Region region, EObject to, IHyperlinkAcceptor acceptor) {
```

Comment:
looks like unwanted api changes to be. maybe adding a second method. doing reflection to find out what is overriden, mark old method deprecated, ....? eclipse api rules are bad

Reply:
@szarnekow any comments / ideas

Reply:
I updated the PR introducing a second method with the `IRegion` parameter and deprecated the method with the `Region` parameter.

Reply:
did you check if there are callers of the method in xtext codebase? and if yes how to ensure calling the old/new method depending on what is overriden by client?

Reply:
I found only one place, the [XtendHyperlinkHelper](https://github.com/eclipse/xtext-xtend/blob/master/org.eclipse.xtend.ide/src/org/eclipse/xtend/ide/hyperlinking/XtendHyperlinkHelper.java#L75)
![screenshot](https://user-images.githubusercontent.com/20393472/61590390-ce995c00-abb8-11e9-91d4-c40e1d5d4f2b.png)
That deprecation can be addressed by declaring/casting the actual parameter from `Region` to `IRegion`. I will address this in a separate PR in the Xtend code base as soon as this one gets merged.